### ISSUE TITLE
Detekt doesn't go along with the new ksp version

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#6612]
## Expected Behavior
When I update kotlin and ksp (I need to upgrade both at the same time) detekt doesn't review new files.

## Observed Behavior
I'm updating my project from kotlin 1.9.10 to 1.9.20 and ksp from `1.9.10-1.0.13` to `1.9.20-1.0.13` (same problem with `1.9.20-1.0.14`). And after this update detekt starts to complain about some files inside the `build/generated` folder.

## Steps to Reproduce
You can reproduce this issue on this project:
https://github.com/BraisGabin/detekt-issue-6612

Clone repo and run `./gradlew detektMain`.

If you downgradle ksp&kotlin to `1.9.10` you will see that everything works fine.

## Context
I added this to my detekt task to debug it:

```kotlin
tasks.withType<Detekt>().configureEach {
    jvmTarget = jdkVersion

    doFirst {
        this as Detekt
        println("@@@@@@@@@@@@@@")
        this.source.visit { println(this) }
        println("@@@@@@@@@@@@@@")
    }
}
```

And before I update the generated files are not listed. As soon as I update kotlin & ksp the generated files appear.

Probably this is not related but while I was researching this I enabled the debug mode on detekt and I found this warnign:

```
warning: classpath entry points to a non-existent location: /Users/brais.gabin/Workspace/project/module/build/generated/ksp/main/classes
```

 I think it is not related because that warning appears with the previous version and the new one.

## Your Environment
* Version of detekt used: 1.23.3
* Version of Gradle used (if applicable): 8.4
* Gradle scan link (add `--scan` option when running the gradle task): https://scans.gradle.com/s/2xfrfh4lugrcw
* Operating System and version: MacOS and Linux (CI)

**Comment by BraisGabin** [comment]
I just find out that this issue is only on the jvm modules. On the android modules everything works as expected.

**Comment by BraisGabin** [comment]
This was really easy to reproduce on a simple project. I updated the first post with a link to the repo with the reproducer and a link to the gradle scan.

**Comment by 3flex** [comment]
Also see https://github.com/google/ksp/issues/1261

I haven't looked at what we can do in our project to work around this yet. Probably just excluding the path(s) in the meantime.

**Comment by BraisGabin** [comment]
Closing as this is a problem on KSP or KGP.
